### PR TITLE
feat(TM-source): add entity data binding provider

### PR DESCRIPTION
**Pull request by sheilaXu**
## Overview
Open source the entity data binding provider from TwinMaker data source. Move some logic from scene-composer into TwinMaker data source.

### Scene Composer
For `scene-composer` package changes specifically, you can preview the component in the published storybook artifact. To do this, wait for the `Publish Storybook` action to complete below.

- Click on the workflow details
- Select the Summary item on the left
- Download the zip file

To run the storybook build locally, you need a local static web server:

```
npm install -g httpserver
cd <Extracted Zip Directory>
httpserver
```

Then open the website http://localhost:8080 to run the doc site.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/source-iottwinmaker/src/data-binding-provider/types.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import { Query, TimeSeriesData } from '@iot-app-kit/core';
+
+export interface ITwinMakerEntityDataBindingContext {
```
Comment (sheilaXu) [inline packages/source-iottwinmaker/src/data-binding-provider/types.ts:3]: most interfaces here are moved from scene-composer with small modification. the copies in scene composer will be removed in next change

File: packages/source-iottwinmaker/src/utils/dataBindingTemplateUtils.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { IDataBindingConfig, IDataFieldOption } from '../data-binding-provider/types';
+
```
Comment (sheilaXu) [inline packages/source-iottwinmaker/src/utils/dataBindingTemplateUtils.ts:2]: moved from scene composer

File: packages/scene-composer/src/components/panels/AddComponentMenu.tsx [hunk 3]
```diff
@@ -128,7 +128,7 @@ export const AddComponentMenu: React.FC<AddComponentMenuProps> = ({ onSelect })
     const component: IEntityBindingComponentInternal = {
       ref: THREE.MathUtils.generateUUID(),
       type: KnownComponentType.EntityBinding,
-      valueDataBinding: { dataBindingContext: '' },
+      valueDataBinding: { dataBindingContext: undefined },
```
Comment (mumanity) [inline packages/scene-composer/src/components/panels/AddComponentMenu.tsx:131]: Why instantiate as `undefined`?

**Review by mumanity — CHANGES_REQUESTED** [review]

File: packages/scene-composer/src/components/panels/AddComponentMenu.tsx (hunk 3 above)
Comment (sheilaXu) [reply packages/scene-composer/src/components/panels/AddComponentMenu.tsx:131]: string is not an expected type for dataBindingContext

**Review by mumanity — APPROVED** [review]

**Pull request merged**